library/database/src/main/java/com/google/android/exoplayer2/database/DatabaseProvider.java

```diff
@@ -50,7 +51,8 @@ public interface DatabaseProvider {
    * need to read from the database.
    *
    * @throws SQLiteException If the database cannot be opened.
+   * @throws Throwable If database inoperable.
    * @return A database object valid until {@link #getWritableDatabase()} is called.
    */
-  SQLiteDatabase getReadableDatabase();
+  SQLiteDatabase getReadableDatabase()  throws Throwable;
```

Comment:
Method getReadableDatabase can throw SQLiteDatabaseLockedException. SQLiteDatabaseLockedException is not inherited from a SQLiteException. So, it won't be handled.

Reply:
https://developer.android.com/reference/android/database/sqlite/SQLiteDatabaseLockedException does extend SQLiteException so it should be handled. Did you hit some issue that it was not?